uint16 encoder1
uint16 encoder2
